core/wiki/macros/toc.tid

```diff
@@ -2,182 +2,212 @@ title: $:/core/macros/toc
         </$button>
       </$reveal>
       <$reveal type="match" stateTitle=<<toc-state>> text="open">
         <$button setTitle=<<toc-state>> setTo="close" class="tc-btn-invisible tc-popup-keep">
-          {{$:/core/images/down-arrow}}
+          {{$:/core/images/down-arrow}}&nbsp;
         </$button>
       </$reveal>
       <<toc-caption>>
     </$link>
     <$reveal type="match" stateTitle=<<toc-state>> text="open">
-      <$macrocall $name="toc-expandable" tag=<<currentTiddler>> sort=<<__sort__>> itemClassFilter=<<__itemClassFilter__>> exclude=<<__exclude__>> path=<<__path__>>/>
+      <$macrocall $name="toc-expandableX" tag=<<currentTiddler>> sort=<<sort>> />
     </$reveal>
     </li>
   </$set>
-</$qualify>
+</$vars>
 \end
 
-\define toc-unlinked-expandable-body(tag,sort:"",itemClassFilter:" ",exclude,path)
-<!-- helper function -->
-<$qualify name="toc-state" title={{{ [[$:/state/toc]addsuffix<__path__>addsuffix[-]addsuffix<currentTiddler>] }}}>
-  <$set name="toc-item-class" filter=<<__itemClassFilter__>> emptyValue="toc-item" value="toc-item-selected">
+\define toc-unlinked-expandable-body()
+\whitespace trim
+<$vars toc-state={{{ [[$:/state/toc]addsuffix[/]addsuffix<name>addsuffix<path>addsuffix[/]addsuffix<currentTiddler>] }}}>
+  <$set name="toc-item-class" filter=<<getItemClass>> emptyValue="toc-item-selected" value="toc-item">
     <li class=<<toc-item-class>>>
       <$reveal type="nomatch" stateTitle=<<toc-state>> text="open">
         <$button setTitle=<<toc-state>> setTo="open" class="tc-btn-invisible tc-popup-keep">
-          {{$:/core/images/right-arrow}}
+          {{$:/core/images/right-arrow}}&nbsp;
           <<toc-caption>>
         </$button>
       </$reveal>
       <$reveal type="match" stateTitle=<<toc-state>> text="open">
         <$button setTitle=<<toc-state>> setTo="close" class="tc-btn-invisible tc-popup-keep">
-          {{$:/core/images/down-arrow}}
+          {{$:/core/images/down-arrow}}&nbsp;
           <<toc-caption>>
         </$button>
       </$reveal>
       <$reveal type="match" stateTitle=<<toc-state>> text="open">
-        <$macrocall $name="toc-expandable" tag=<<currentTiddler>> sort=<<__sort__>> itemClassFilter=<<__itemClassFilter__>> exclude=<<__exclude__>> path=<<__path__>>/>
+        <$macrocall $name="toc-expandableX" tag=<<currentTiddler>> sort=<<sort>> />
       </$reveal>
     </li>
   </$set>
-</$qualify>
+</$vars>
 \end
 
 \define toc-expandable-empty-message()
-<$macrocall $name="toc-linked-expandable-body" tag=<<tag>> sort=<<sort>> itemClassFilter=<<itemClassFilter>> exclude=<<excluded>> path=<<path>>/>
+<$macrocall $name="toc-linked-expandable-body"/>
 \end
 
-\define toc-expandable(tag,sort:"",itemClassFilter:" ",exclude,path)
-<$vars tag=<<__tag__>> sort=<<__sort__>> itemClassFilter=<<__itemClassFilter__>> path={{{ [<__path__>addsuffix[/]addsuffix<__tag__>] }}}>
-  <$set name="excluded" filter="""[enlist<__exclude__>] [<__tag__>]""">
+\define toc-expandableX(tag,sort:"")
```

Comment:
This seems an unexpected name for a macro because it doesn't meet the conventions of all the other macro names. Are you trying to mark it as internal or private? If so, I'd prefer to do it unambiguously: `\define toc-expandable_private(...` etc.

Reply:
Yea. It's a wrapper, that was needed, to handle the "exclude" parameter in the right way. Also changing from `"""$var$"""` to `<<___var__>>` forced some changes in the logic. ...

I use the capital X for extra ... BUT I'll rename it to `_private`